packages/core/src/internal/channel.js

```diff
@@ -116,15 +121,15 @@ export function eventChannel(subscribe, buffer = buffers.none()) {
 
   unsubscribe = subscribe(input => {
     if (isEnd(input)) {
-      close()
       closed = true
```

Comment:
How about moving this statement (`closed = true`) inside `close()`?  

`multicastChannel()` and `channel()` have this statement inside their `close()` too.